### DISCUSSION TITLE
Migration from CommandSchedulerBundle

### DISCUSSION THREAD

**Discussion by chiqui3d** [Guikingone/SchedulerBundle#206] [Q&A] [answered]
Hello,

First thanks for this awesome Bundle, damn 😃, it has so many options, so I'm going for the easy stuff first. Currently, I'm using `CommandSchedulerBundle `and I want to change Bundle as this is not maintained, with this Bundle I have an admin panel that I add to the database and then consume them without any problem.

I understand that adding https://github.com/Guikingone/SchedulerBundle/blob/main/doc/transport.md#doctrine with `auto_setup to false` and adding new tasks from the admin I will not have any problem, right?

**Comment by Guikingone** [comment, marked as answer]
Hi @chiqui3d 👋🏻 

Yes, the `auto_setup` option can be set if you don't use the migration that the bundle can generate for you, if the table hasn't been created, the Doctrine connection will do it BEFORE triggering an operation, be aware that even if the performances impact still minimal, it can slow down the first query 🙂

**Comment by chiqui3d** [comment]
I'm getting a bit confused, as I see that it is possible for me to overwrite the tasks I create from my panel and add them directly to the schedule table. It is not possible to get the tasks from a class directly without having to configure it in the tasks config of the yaml file, something like this:

```
<?php

namespace App\Scheduler;

use SchedulerBundle\SchedulerInterface;
use SchedulerBundle\Task\CommandTask;

final class ScheduleCommand
{
    public function __invoke(SchedulerInterface $scheduler, TaskRepository $taskRepository): void
    {
        $tasks = $taskRepository->findAll();
        foreach ($tasks as $task){
            $scheduler->schedule(new CommandTask($task->getName(), $task->getCommand()));
        }

    }
}
```

**Reply by Guikingone** [reply]
I'm not getting the point here, sorry 🤔 

You can configure tasks using the configuration and / or using the `SchedulerInterface::schedule()` method, can you be more precise about the issue ? 🤔

**Reply by chiqui3d** [reply]
Sorry, I think I understand a little better.  It seems that any task you add by default runs every minute, doesn't it, or is there an option that says otherwise?

I have also used the following in MyController to save a command and I get the following error:

```
$scheduler->schedule(new CommandTask('test-command', 'cache:clear',[],[
            'description' => 'Esto es un test',
            "expression" => '30 20 * * *',
            "timezone" => 'Europe/Madrid',
            'max_executions' => null,
            'output' => true
        ]));
```

`The option "options" with value array is expected to be of type "string[]", but one of the elements is of type "null".
`

**Reply by chiqui3d** [reply]
It seems that the options parameter are not for that. I don't understand how to configure the Timezone, expression, etc...

**Reply by Guikingone** [reply]
Oh, I get it, you're trying to configure the "internal" task options (which is not possible using constructor arguments of `CommandTask`, the third argument is the options for `cache:clear` (like `env`, `verbosity` and so on). 

If you need to configure the internal options, you must use the defined setters 🙂
